lib/modelinstance.js

```diff
@@ -212,7 +212,9 @@ function _encodeValue(context, type, value, forceTyping, f) {
     if (!(value instanceof ModelInstance)) {
       throw new Error('Expected ' + f.name + ' type to be a ModelInstance.');
     }
-    if (type.name !== value.$.schema.name) {
+    // Values must match stated type names, unless the reference is to
+    // 'Mixed', then any reference will do.
+    if (type.name !== value.$.schema.name && (type.name !== 'Mixed')) {
```

Comment:
Did some more research.  Not a bug so much as a design oddity.

[In ModelRefs here](https://github.com/couchbaselabs/node-ottoman/blob/master/lib/schema.js#L32) you'll notice that they're created with a "name" attribute to refer to the type of model.  The core type `MixedType` in the end is a data structure that is `{type: "Mixed"}`.  So they're really not the same, because a ModelRef in this case ends up with `name: "Mixed"`.  It is indeed a type of ModelRef, but since ModelRef chose to store "name" for the model reference and not "type", when we extended this to do mixed references, the two don't match.   

So basically I think I need to stand by this code as it's written.  A solution could be to change ModelRefs to store a "type" instead of a "name" but I don't know what else depends on that, and it would seem likely to cause bigger problems than it solves.